app/models/application_record.rb

```diff
@@ -50,6 +50,10 @@ def add_response_data(prompt, response)
     update(data: data << [prompt, response])
     self
   end
+  
+  def questions_hash
+    data.map{|arr| [arr[3], arr[1]]}.select{|arr| [arr[0], arr[1]].all?}.to_h
```

Comment:
Elongated in the sense of tabbing it to multiple lines and commenting it or what? Not entirely sure how I'd extend the code itself.